source/common/http/conn_manager_impl.cc

```diff
@@ -1165,6 +1165,23 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::encodeMetadata(ActiveStreamEncoderFilter* filter,
+                                                         MetadataMap& metadata_map) {
+  resetIdleTimer();
+
+  std::list<ActiveStreamEncoderFilterPtr>::iterator entry = commonEncodePrefix(filter, false);
+  for (; entry != encoder_filters_.end(); entry++) {
+    // TODO(soya3129): Add filters here.
+  }
+
+  ENVOY_STREAM_LOG(debug, "encoding metadata via codec:\n", *this);
```

Comment:
Also for your mental TODO list I'd like to have a iostream function on metadata so we can print key value pairs for incoming and outgoing metadata.
No action needed on this PR but maybe the next?

Reply:
Good call! Added an iostream function.